### PR TITLE
debug: rename begin_debug_fmt and add non-newline version

### PR DESCRIPTION
**Pull request by jettr**
Remove `begin` prefix from `begin_debug_fmt` and `begin_debug_verbose_fmt` since there does not appear to be a finish or end that should accompany it.

Also add a non-newline version of this function.

This does not need to make it in by Tock 2.0 release

### Documentation Updated

- None

### Formatting

- [x] Ran `make prepush`.

### CODE REVIEW

**Review by alistair23 — DISMISSED** [review]

**Review by hudson-ayers — DISMISSED** [review]

File: kernel/src/debug.rs [hunk 1]
```diff
@@ -527,23 +527,45 @@ impl Write for DebugWriterWrapper {
...
+pub fn debug_format(args: Arguments) {
     let writer = unsafe { get_debug_writer() };
 
     let _ = write(writer, args);
-    let _ = writer.write_str("\r\n");
     writer.publish_bytes();
 }
 
-pub fn begin_debug_verbose_fmt(args: Arguments, file_line: &(&'static str, u32)) {
+pub fn debug_formatln(args: Arguments) {
```
Comment (ppannuto) [inline kernel/src/debug.rs:537]:
```suggestion
pub fn debug_println(args: Arguments) {
```

If we're going to rename, I think this makes more sense -- I think I would expect a `formatln` to just write bytes into a buffer; this does the print.

File: kernel/src/debug.rs [hunk 2]
```diff
@@ -527,23 +527,45 @@ impl Write for DebugWriterWrapper {
...
 
     let (file, line) = *file_line;
     let _ = writer.write_fmt(format_args!("TOCK_DEBUG({}): {}:{}: ", count, file, line));
     let _ = write(writer, args);
+}
+
+pub fn debug_verbose_format(args: Arguments, file_line: &(&'static str, u32)) {
+    let writer = unsafe { get_debug_writer() };
+
+    print_debug_header(writer, args, file_line);
```
Comment (ppannuto) [inline kernel/src/debug.rs:561]:
```suggestion
    print_debug_header(writer, file_line);
    let _ = write(writer, args);
```

Feels like this part should be here (or the "print header" function should be renamed to something like `debug_println_with_header`)

**Review by ppannuto — CHANGES_REQUESTED** [review]
Overall looks fine -- I think the `begin` is a legacy of a long-dead implementation of debug.

Some clarification comments on the naming that I think/hope makes things a bit clearer

File: kernel/src/debug.rs (hunk 2 above)
Comment (jettr) [reply kernel/src/debug.rs:561]: Ah yes, I was trying to dedup that extra line, but then the function name gets tricky. I have moved the arg print back into both functions and updated the function names. Thanks!

File: kernel/src/debug.rs (hunk 1 above)
Comment (jettr) [reply kernel/src/debug.rs:537]: Good call. Done.

**Review by hudson-ayers — APPROVED** [review]

**Review by ppannuto — APPROVED** [review]
bors d+

**Comment by ppannuto** [comment]
Just blocked on 2.0

**Review by lschuermann — APPROVED** [review]

**Review by alistair23 — APPROVED** [review]

**Pull request merged**